packages/core/src/utils.ts

```diff
@@ -741,3 +743,76 @@ function toPreciseInteger(a: number, precision?: number): number {
   }
   return Math.round(a * Math.pow(10, precision));
 }
+
+interface Hl7DateParseOptions {
+  seconds?: boolean;
+  tzOffset?: string;
+}
+
+/**
+ * Returns a formatted string representing the date in ISO-8601 format.
+ * @param hl7Date Date string.
+ * @param options Optional configuration Object
+ * @returns
+ */
+export function parseHl7Date(hl7Date: string | undefined, options?: Hl7DateParseOptions): string | undefined {
+  if (!hl7Date) {
+    return undefined;
+  }
+
+  options = { ...{ seconds: true, tzOffset: 'Z' }, ...options };
+
+  const year = Number.parseInt(hl7Date.substring(0, 4));
+  const month = Number.parseInt(hl7Date.substring(4, 6));
+  const date = Number.parseInt(hl7Date.substring(6, 8));
+  const hours = Number.parseInt(hl7Date.substring(8, 10));
+  const minutes = Number.parseInt(hl7Date.substring(10, 12));
+
+  const seconds = options.seconds ? Number.parseInt(hl7Date.substring(12, 14)) : 0;
+
+  return `${pad2(year)}-${pad2(month)}-${pad2(date)}T${pad2(hours)}:${pad2(minutes)}:${pad2(seconds)}.000${
+    options.tzOffset
+  }`;
+}
+
+function pad2(n: number): string {
+  return n.toString().padStart(2, '0');
+}
+
+/**
+ * Finds the first observation in the input array that matches the specified code and system.
+ * @param observations - The array of observations to search.
+ * @param code - The code to search for.
+ * @param system - The system to search for.
+ * @returns The first observation in the input array that matches the specified code and system, or undefined if no such observation is found.
+ */
```

Comment:
I think this is a good function to have, but it should be generalized to more resource types than just `Observations`. 
- Any resourceType that has a `code` property of type `CodeableConcept` or `CodableConcept[]`
- Allow `code` param to be a string or `CodeableConcept`.